Source/WebGPU/WebGPU/Buffer.swift

```diff
@@ -23,7 +23,24 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-public func copySpan(destination: SpanUInt8, source: SpanConstUInt8) {
-    precondition(source.size_bytes() <= destination.size_bytes(), "Destination buffer not big enough.")
-    memcpy(destination.__dataUnsafe(), source.__dataUnsafe(), source.size_bytes())
+import WebGPU_Internal
+
+extension WebGPU.Buffer {
+    var bufferContents: UnsafeMutableRawBufferPointer {
+        UnsafeMutableRawBufferPointer(start: m_buffer.contents(), count: m_buffer.length)
+    }
+
+    func copy(from data: SpanConstUInt8, offset: Int) {
+        let slice = bufferContents[offset...]
+        // copyBytes(from:) checks bounds in debug builds only.
+        // FIXME: Use a bounds-checking implementation when one is available.
+        precondition(slice.count <= data.size_bytes())
+        slice.copyBytes(from: data)
+    }
+}
+
+// FIXME(emw): Find a way to generate thunks like these, maybe via a macro?
```

Comment:
We could generate these, but only if we dropped the name prefix on all function arguments.

Reply:
I'll post my macro implementation next week — it generates working code, though it doesn't get emitted to the C++ header due rdar://139984562.